src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +47,7 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
-                a = DNS_PACKET_SIZE_START;
-        else
-                a = mtu - UDP_PACKET_HEADER_SIZE;
-
-        if (a < DNS_PACKET_HEADER_SIZE)
-                a = DNS_PACKET_HEADER_SIZE;
+        a = MAX(mtu, DNS_PACKET_HEADER_SIZE);
```

Comment:
Yes, indeed. But it took me quite a while to actually grok what the old code was supposed to do... I stared for 20min straight on your patch trying to figure out wtf the purpose of the old code was and then I got it... Sorry for not spending those 20min back then!

(might be indication to add a comment, so that the next time it's not that hard to grok again)